packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -223,6 +224,19 @@ export const BootDataProvider = ({
     setCachedBootData((cachedData) => updateLocalBootData(cachedData, { exp }));
   }, []);
 
+  useEffect(() => {
+    const partialUser = user as Partial<LoggedUser>;
+
+    gqlClient.setHeader(
+      'content-language',
+      partialUser?.language || ContentLanguage.English,
+    );
+
+    return () => {
+      gqlClient.setHeader('content-language', ContentLanguage.English);
+    };
```

Comment:
This sets the language globally for all gql requests made with a client.

Reply:
But why in a useEffect?
Can't we solve it without just on the fact if `provider in user` or something?

Just want to avoid 1000s of rerenders on bootprovider where possible.

Reply:
Its no longer in useEffect, check again. I did it inline because useEffect would be too late for extension.

Reply:
https://github.com/dailydotdev/apps/pull/3469#discussion_r1725091836